src/app/static/src/js/components/vuex/runReport/gitSelections.vue

```diff
@@ -0,0 +1,36 @@
+<template>
+    <div>
+        <div v-if="metadata && metadata.git_supported" id="git-branch-form-group" class="form-group row">
+            <label for="git-branch" class="col-sm-2 col-form-label text-right">Git branch</label>
+            <div class="col-sm-6">
+                <select id="git-branch" class="form-control">
+                    <option v-for="branch in gitBranches" :key="branch" :value="branch">
+                        {{ branch }}
+                    </option>
+                </select>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+    import Vue from "vue";
+    import {mapState} from "vuex";
+    import {RunReportRootState} from "../../../store/runReport/store";
+    import {EmptyObject, RunReportMetadataDependency} from "../../../utils/types";
+
+    interface Computed {
+        metadata: RunReportMetadataDependency
+        gitBranches: string[]
+    }
+
+    export default Vue.extend<EmptyObject, EmptyObject, Computed, EmptyObject>({
+        name: "GitSelections",
+        computed: {
+            ...mapState({
+                metadata: (state: RunReportRootState) => state.git.metadata,
```

Comment:
Could just map to `GitState` and namespace the `mapState` call?

Reply:
is that better? I don't love referencing namespaces as strings, although I guess we already have to do that for actions/mutations

Reply:
Yeah, I think we've tended to scope the components to the module namespace where we can in the past, though I guess it doesn't gain us very much, just makes the `mapState` a little terser (usually). It's also made components feel consistent with actions/mutations which are generally scoped to module states - but we've seen that consistency break down anyway with components which use multiple states. So maybe not much value in it.

If we did specify the module, I guess we wouldn't need to use string literals  at least, could keep those as consts somehwhere.

Reply:
in the next PR there will be actions/mutations mapped which def do need the namespace so think you're right, may as well be consistent and re-use the namespace. this particular component might actually map exactly onto the `git` module. nice idea to put the consts in a shared location as well, will do that :+1: